.travis.yml

```diff
@@ -2,43 +2,59 @@ language: node_js
 node_js: 
   - "8"
   - "10"
+  - "12"
 
 script: npm run unit-test
 
+common: &integration-common
+  script:
+    - npm run build
+    - node build_release_package_json.js
+    - npm run integration-test
+  node_js: "8"
+
 jobs:
   include:
-    - stage: build
+    - stage: lint
+      script: npm run lint
+    - &building
+      stage: build
       script: npm run build
```

Comment:
you can extract the stage and script property into the `common:` definition
```
common:
 buildstage: &build
  stage: build
  script: npm run build

...

-stage <<: *build
 node_js: 8
-stage <<: *build
 node_js: 10
-stage <<: *build
 node_js: 12

```

Reply:
The benefit is no default values for node_js so we can be explicit that the difference between these stage definition is the node versions